File: problems/0567-permutation-in-string/python/solution.py
def checkInclusion(s1, s2):
    """Permutation in String — https://leetcode.com/problems/permutation-in-string/"""
    if len(s1) > len(s2):
        return False
    need = [0] * 26
    window = [0] * 26
    for c in s1:
        need[ord(c) - 97] += 1
    for i, c in enumerate(s2):
        window[ord(c) - 97] += 1
        if i >= len(s1):
            window[ord(s2[i - len(s1)]) - 97] -= 1
        if window == need:
            return True
    return False

File: problems/0567-permutation-in-string/go/solution.go
package solution

// Permutation in String — https://leetcode.com/problems/permutation-in-string/
func checkInclusion(s1, s2 string) bool {
	if len(s1) > len(s2) {
		return false
	}
	var need, window [26]int
	for i := 0; i < len(s1); i++ {
		need[s1[i]-'a']++
	}
	for i := 0; i < len(s2); i++ {
		window[s2[i]-'a']++
		if i >= len(s1) {
			window[s2[i-len(s1)]-'a']--
		}
		if window == need {
			return true
		}
	}
	return false
}
